backends/ref/ceed-ref-operator.c

```diff
@@ -662,9 +658,8 @@ static int CeedOperatorSetupFieldsAtPoints_Ref(CeedQFunction qf, CeedOperator op
             CeedBasisApplyAtPoints(basis, max_num_points, CEED_NOTRANSPOSE, CEED_EVAL_WEIGHT, CEED_VECTOR_NONE, CEED_VECTOR_NONE, q_vecs[i]));
         break;
     }
-    if (is_input && e_vecs[i]) {
-      CeedCallBackend(CeedVectorSetArray(e_vecs[i], CEED_MEM_HOST, CEED_COPY_VALUES, NULL));
-    }
+    if (e_vecs[i]) CeedCallBackend(CeedVectorSetValue(e_vecs[i], 0.0));
+    if (eval_mode != CEED_EVAL_WEIGHT) CeedCallBackend(CeedVectorSetValue(q_vecs[i], 0.0));
   }
```

Comment:
More learning moment for me, why the shift of these operations downwards?

Reply:
Ok,, I see why `q_vecs` got moved. But not sure about the change to `e_vecs`

Reply:
I realized we had `CeedVectorSetValue` separately in all but 1 case, so I wanted the initialization logic together.

`CeedVectorSetArray` with a `NULL` array and `CeedVectorSetValue` with `0.0` both initialize the memory for the vector. We were using both in this function so I went with `CeedVectorSetValue` as it felt clearer what was happening than `CeedVectorSetArray` with `NULL`.

Reply:
I updated the usages to match - interestingly only the CPU CeedOperator impls seemed to do this.